### DISCUSSION TITLE
Run open-vmware-tools on talos?

### DISCUSSION THREAD

**Discussion by ghost** [siderolabs/talos#3649] [Q&A]
I'm trying to setup [vsphere-csi-driver](https://vsphere-csi-driver.sigs.k8s.io/driver-deployment/installation.html#vsphereconf_for_block) on a talos cluster running inside vmware esxi (with vcenter ofc). The main issue I'm running into is that the "cloud-controller" cannot find my vms by ip/dns, and this information actually isn't available in vcenter since these vms don't have vmware-tools.

Has anyone found a workaround for this? I've seen some ideas about possibly deploying a privileged container whose image has open-vmware-tools built-in on the host-network of each host; however, I don't see any recent builds like this. Most of these solutions look like they're 4-5 years old.

**Comment by andrewrynhard** [comment]
@alex1989hu might have some ideas.

**Reply by alex1989hu** [reply]
Hello, we have been using for years:
<https://github.com/linuxkit/linuxkit/tree/master/contrib/open-vm-tools>

**Reply by ghost** [reply]
Oh interesting. I will give that a try!
